test/test_compatibility_ext_and_com.py

```diff
@@ -0,0 +1,89 @@
+from numpy.testing import assert_array_almost_equal as equal
+
+import pypsa
+from pypsa.descriptors import nominal_attrs
+
+
+def test_compatibility_ext_and_comt():
```

Comment:
It would be good to add a case copy using extendable on and commitable with p_nom = p_nom_mod.
This means that the solution should be equivalent to the one of a standard unit commitment, without extendability being on.
As template, you could use one test of unit commitment.
What do you think?

Reply:
@cfvescovo and @LoZack19 told me that they would like to give a help with this PR, designing together a more appropriate test, since I would like to improve the current one, which is not so challenging.

I got the point, but obiusly keeping extendability the results won't be comparable with actual "unit commitment test". maybe we could add a case like the one you suggested directly in the network proposed as test. What do you think?

Reply:
> @cfvescovo and @LoZack19 told me that they would like to give a help with this PR, designing together a more appropriate test, since I would like to improve the current one, which is not so challenging.

Chiming in to confirm my availability.
W.r.t. the functions naming/duplication issue, I think a minimalist approach should be followed. Creating 12 functions to handle separately every single case should be avoided.

Reply:
Another point to consider: is backwards compatibility crucial for a library which hasn't reached API stability (1.0) yet?
IMHO cleaner code and a simpler API should be preferred over backwards compatibility.

Disclaimer: I am not a maintainer so take this with a grain of salt.

Reply:
I agree with @cfvescovo about the need to create clean code, especially at this stage of pypsa development.